components/org.wso2.carbon.identity.verification.onfido.connector/src/main/java/org/wso2/carbon/identity/verification/onfido/connector/internal/OnfidoIDVDataHolder.java

```diff
@@ -26,20 +26,19 @@
  */
 public class OnfidoIDVDataHolder {
 
-    public static OnfidoIDVDataHolder instance = new OnfidoIDVDataHolder();
```

Comment:
Previously we have used a singleton for `OnfidoIDVDataHolder` class. Is there a particular reason to remove it?

Reply:
The decision to remove the singleton pattern from the `OnfidoIDVDataHolder` class and make the methods static was driven by the fact that the attributes `realmService` and `identityVerificationManager` were already static. Since these static attributes belong to the class itself, using a singleton instance to access them was redundant. The class does not manage any state that requires instantiation, so making the methods static simplifies access, reduces unnecessary complexity, and aligns the design with the class's actual usage as a utility for accessing static services.